docs/dependency_confusion.md

```diff
@@ -61,80 +61,78 @@ to verify that the content of all direct and indirect dependencies is what you e
+tell Cachito to process them.
+
+Cachito further restricts what you can put in your requirements.txt files. All of the dependencies
+must be
+[pinned](https://github.com/release-engineering/cachito/blob/master/docs/pip.md#pinning-versions)
+to an exact version. Cachito will refuse to process requirements files that use the --index-url or
+--extra-index-url options, which means private registries are out of the question. These two
+restrictions should eliminate most attack vectors.
+
+To protect yourself even further, use pip’s
+[hash-checking mode](https://pip.pypa.io/en/stable/reference/pip_install/#hash-checking-mode). Note
+that pip does not support hash checking for VCS dependencies, e.g. git. Consider transforming your
+git dependencies to plain https dependencies. For example, if the repository is hosted on github,
+you can use https://github.com/{org_name}/{repo_name}/{commit_id}.tar.gz to get the tarball for a
+specific commit.
+
+## npm
+
+*TL;DR: do not use unofficial registries. Even if you try to do so via .npmrc, Cachito will ignore
+it.*
+
+Npm packages follow the typical package file + lock file approach. The package file is
+[package.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-json), the lock file is
+[package-lock.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json) or
+[npm-shrinkwrap.json](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json). Cachito
+requires the lock file.
+
+The lock file pins all dependencies to exact versions. For https dependencies, which are impossible
+to pin, Cachito requires the integrity value (a checksum). For dependencies from the npm registry,
+Cachito does not require integrity values, but newer versions of npm will always include them. Check
+if all your dependencies have an integrity value, update your lock file if not.
+
+Do not try to use private registries. If you point npm to a private registry (or any registry other
+than the official one) via [.npmrc](https://docs.npmjs.com/cli/v6/configuring-npm/npmrc), Cachito
+will ignore it and look in the official registry anyway.
+
+## yarn
+
+*TL;DR: same as npm, but the handling of unofficial registries is slightly less surprising. Still,
+you probably should not use them.*
+
+Yarn packages are identical to npm packages except that they use
+[yarn.lock](https://classic.yarnpkg.com/en/docs/yarn-lock/) as the lock file. Everything that
+applies to npm applies to yarn.
+
+The one difference is the handling of unofficial registries. If you point yarn to an unofficial
+registry via .npmrc or [.yarnrc](https://classic.yarnpkg.com/en/docs/yarnrc), this will be reflected
+in the resolved url in the lock file. Cachito will see that the url does not point to the official
+registry and will treat the dependency as a plain https dependency. If the url is accessible to
+Cachito, it will download the dependency directly without relying on npm/yarn dependency resolution.
```

Comment:
While this is probably true for yarn v1 which cachi2 can't do yet, it is definitely not true for v3 where we solely rely on yarn's native mechanism to download deps.